sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/repository/support/SimpleCosmosRepository.java

```diff
@@ -45,6 +46,32 @@ public SimpleCosmosRepository(CosmosEntityInformation<T, ID> metadata,
         if (this.information.isAutoCreateContainer()) {
             createContainerIfNotExists();
         }
+
+        CosmosContainerProperties currentProperties = getContainerProperties();
+        if (shouldUpdateIndexingPolicy(currentProperties)) {
+            currentProperties.setIndexingPolicy(information.getIndexingPolicy());
+            replaceContainerProperties(currentProperties);
+        }
+    }
+
+    private boolean shouldUpdateIndexingPolicy(CosmosContainerProperties currentProperties) {
+        return currentProperties != null && !currentProperties.getIndexingPolicy().equals(information.getIndexingPolicy());
```

Comment:
i wanted to go ahead and open this PR to start discussion of this line. The issue here is that the indexPolicy object retrieved from the server will not ever compare with the index policy that is contained in CosmosEntityInformation using .equals(). This is for a few reasons....
1.) The IndexPolicy object only contains a few instance fields and the rest of the data is contained in the serialized version of the object that is contained internally. The version of indexPolicy that is within CosmosEntityInformation will never contain a serialized version of the object that is comparable to the one retrieved from the server due to all the metadata fields that are added to the serialized json by the server.
2.) The version of the indexPolicy object in CosmosEntityInformation is created from the parameters provided by the annotation. However the annotation does not account for any of the implied state that is set in certain cases. For instance, if you apply the CosmosIndexingPolicy annotation with no parameters, then the default indexing policy will get applied to your container. The index policy object in CosmosEntityInformation will contain an empty list for the included and excluded fields in this case. The index policy object retrieved from the server will contain an included list with "/*"  and an excluded list with "/_etag/*".

Since the way the index policy is updated is by doing a container replace, it seems like this check needs to be accurate as far as detecting if the index policy actually changed so that needless container replaces are not being done.

The only approach i can think of to do this is to account for all the discrepancies between these objects when comparing.

Can you guys think of a better way of determining if an index policy update is needed?

Reply:
@kushagraThapar any thoughts on this? The way i see it I either need to flesh out the comparison code to account for the discrepancies mentioned above or we omit the check entirely and let the container replace run everytime. Do you know what the cost of a container replace is? If it is an inexpensive constant time operation then letting it always run is probably fine. If it is something that is expensive and scales with the size of the container then we should ensure that needless container replaces are not being executed.

Reply:
@Blackbaud-EricSlater - It is an expensive call that we should not do all the times. The current equals code that you have on this PR is costly or is it fine ?

Reply:
its not that the equals comparison is costly, its that it never returns true. I am working on fixing this now. I am adding definitions for .equals to the IncludedPath, ExcludedPath and Index classes and adding some additional code that accounts for the discrepancies i mentioned above.

Reply:
Makes sense, that will be the right way to fix this.

Reply:
this is fixed now and the logic for comparing indexPolicies is contained in IndexPolicyCompareService. Implementing .equals on ExcludedPath and IncludedPath was necessary as well